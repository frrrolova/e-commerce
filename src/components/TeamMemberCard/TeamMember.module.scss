@import '../../styles/variables.scss';

.container {
  position: relative;
  flex-grow: 1;
  flex-basis: 0;

  .card {
    border-radius: $border-radius-card;
    opacity: $opacity-item;
    display: flow-root;
    height: 1100px;
    @media screen and (max-width: 899px){
      height: auto;
    }

    &::before {
      content: "";
      position: absolute;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      border-radius: $border-radius-card;
      border: 2px solid transparent;
      background: linear-gradient(45deg, $border-dark-color, $border-light-color, $border-dark-color) border-box;
      -webkit-mask: linear-gradient(#fff 0 0) padding-box, linear-gradient(#fff 0 0);
      -webkit-mask-composite: destination-out;
      mask-composite: exclude;
    }
  }
}

.inner {
  min-width: 400px;
  @media screen and (max-width: 1380px){
    min-width: 300px;
  }
  @media screen and (max-width: 980px){
    min-width: 250px;
  }
}

.img {
  height: 400px;
  min-width: 300px;
  width: 300px;
  @media screen and (max-width: 1150px){
    height: 300px;
    min-width: 250px;
    width: 250px;
  }
  @media screen and (max-width: 1000px){
    height: 250px;
    min-width: 200px;
    width: 200px;
  }
  @media screen and (max-width: 899px){
    min-width: 600px;
    width: 600px;
    height: 600px;
  }
  @media screen and (max-width: 680px){
    min-width: 500px;
    width: 500px;
    height: 500px;
  }
  @media screen and (max-width: 600px){
    min-width: 400px;
    width: 400px;
    height: 400px;
  }
  @media screen and (max-width: 480px){
    min-width: 300px;
    width: 300px;
    height: 300px;
  }
  @media screen and (max-width: 370px){
    min-width: 250px;
    width: 250px;
    height: 250px;
  }
  @media screen and (max-width: 325px){
    min-width: 200px;
    width: 200px;
    height: 200px;
  }
}

.bio {
  height: 170px;
  @media screen and (max-width: 1205px){
    height: 200px;
  }
  @media screen and (max-width: 1065px){
    height: 250px;
  }
  @media screen and (max-width: 985px){
    height: 300px;
  }
  @media screen and (max-width: 899px){
    height: auto;
  }
}
