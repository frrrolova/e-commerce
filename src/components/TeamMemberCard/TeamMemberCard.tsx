import { Box, Link, Paper, Typography } from '@mui/material';
import { TeamMember } from '@/types';
import '@/styles/styles.scss';
import styles from './TeamMember.module.scss';
// import CheckIcon from '@mui/icons-material/Check';

interface TeamMemberCardProps {
  teamMember: TeamMember;
}

function TeamMemberCard({ teamMember }: TeamMemberCardProps) {
  return (
    <>
      <Box className={styles.container}>
        <Paper
          className={styles.card}
          elevation={3}
          sx={{
            backgroundColor: 'transparent',
          }}
        >
          <Box
            component="div"
            sx={{
              minWidth: {
                xs: '300px',
                md: '200px',
                lg: '300px',
              },
              display: 'flex',
              flexDirection: 'column',
              gap: '10px',
              padding: '20px',
            }}
          >
            <Box
              component="header"
              sx={{
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'space-between',
                borderBottom: '1px solid gray',
                p: 1,
              }}
            >
              <Typography component="h2">{teamMember.name}</Typography>
              <Link
                sx={{
                  position: 'relative',
                  zIndex: '10',
                }}
                href={teamMember.git}
              >
                Github
              </Link>
            </Box>
            <Box
              component="div"
              sx={{
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'center',
                p: 1,
              }}
            >
              <Box
                sx={{
                  width: {
                    xs: '400px',
                    md: '200px',
                    lg: '300px',
                  },
                  minWidth: {
                    xs: '400px',
                    md: '200px',
                    lg: '300px',
                  },
                  height: {
                    xs: '400px',
                    md: '200px',
                    lg: '300px',
                  },
                  borderRadius: '50%',
                  backgroundImage: `url('${teamMember.photo}')`,
                  backgroundRepeat: 'no-repeat',
                  backgroundSize: 'cover',
                  backgroundPosition: 'center center',
                  overflow: 'hidden',
                }}
              ></Box>
              <Typography
                sx={{
                  fontStyle: 'italic',
                  textAlign: 'center',
                }}
                component="p"
              >
                {teamMember.role}
              </Typography>
              <Typography
                component="h4"
                sx={{
                  alignSelf: 'flex-start',
                  fontSize: '18px',
                  marginY: '10px',
                  fontWeight: 'bold',
                }}
              >
                Bio
              </Typography>
              <Typography
                sx={{
                  height: {
                    xs: 'auto',
                    md: '220px',
                    lg: '170px',
                  },
                }}
                component="h5"
              >
                {teamMember.bio}
              </Typography>
            </Box>
            <Box component="footer">
              <Typography
                component="h3"
                sx={{
                  alignSelf: 'flex-start',
                  fontSize: '18px',
                  marginY: '10px',
                  fontWeight: 'bold',
                }}
              >
                Contributions
              </Typography>
              <Box component="div">
                {teamMember.contributions.map((item, index) => (
                  <Box
                    sx={{
                      fontSize: '14px',
                      p: 1,
                      borderBottom: '1px solid gray',
                      '&:last-child': { borderBottom: 'none' },
                    }}
                    key={index}
                  >
                    âœ… {item}
                  </Box>
                ))}
              </Box>
            </Box>
          </Box>
        </Paper>
      </Box>
    </>
  );
}

export default TeamMemberCard;
