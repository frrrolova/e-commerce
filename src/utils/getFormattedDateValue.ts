export function getFormattedDateValue(value: string) {
  const date = new Date(value);
  const dateValue: string = new Date(date.getTime() - date.getTimezoneOffset() * 60000).toISOString().substring(0, 10);
  return dateValue;
}
