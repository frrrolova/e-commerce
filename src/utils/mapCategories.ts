import { Category } from '@/types';
import { Category as CategorySDK } from '@commercetools/platform-sdk';

export function mapCategories(categoryResponse: CategorySDK[]): Category[] {
  return categoryResponse.map((category) => ({
    id: category.id,
    name: category.name['en-US'],
    key: category.key,
    ancestors: category.ancestors,
    parent: category.parent,
  }));
}
