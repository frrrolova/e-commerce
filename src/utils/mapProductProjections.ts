import { Product } from '@/types';
import { ProductProjection } from '@commercetools/platform-sdk';

export function mapProductProjections(projections: ProductProjection[]): Product[] {
  return projections.map((projection) => ({
    id: projection.id,
    name: projection.name['en-GB'],
    description: projection.description?.['en-GB'] || 'Product has no description',
    images: projection.masterVariant.images,
    prices: projection.masterVariant.prices,
  }));
}
