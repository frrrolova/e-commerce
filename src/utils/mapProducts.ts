import { Product as ProductCT } from '@commercetools/platform-sdk';
import { Product } from '@/types';

// Map the commercetools products to UI-product type
export const mapProducts = (prodactsCT: ProductCT[]): Product[] => {
  const products: Product[] = prodactsCT.map((product: ProductCT) => {
    return {
      id: product.id,
      name: product.masterData.current.name[`en-GB`],
      description: product.masterData.current.description?.[`en-GB`] || 'Product has no description',
      images: product.masterData.current.masterVariant.images,
      prices: product.masterData.current.masterVariant.prices,
    };
  });
  return products;
};
