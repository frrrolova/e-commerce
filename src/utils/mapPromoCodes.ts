import { PromoData } from '@/types';
import { CartDiscount } from '@commercetools/platform-sdk';

export function mapPromoCodes(promoResponse: CartDiscount[]): PromoData[] {
  return promoResponse.map((code, index) => ({
    id: code.id,
    heading: code.name['en-GB'],
    imgPath: `/images/home/promo-${index}.webp`,
    description: code.description ? code.description['en-GB'] : '',
    subHeading: code.key || '',
  }));
}
