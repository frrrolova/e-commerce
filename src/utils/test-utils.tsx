import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';

import type { AppStore, RootState } from '../store/store';
// As a basic setup, import your same slice reducers
import initStore from '../store/store';
import { MemoryRouter } from 'react-router-dom';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { Project } from '@commercetools/platform-sdk';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

const mockShop: Project = {
  countries: [],
  createdAt: new Date().toISOString(),
  languages: [],
  key: 'test-key',
  version: 2,
  name: 'Plant Shop',
  currencies: [],
  messages: { enabled: false, deleteDaysAfterCreation: 15 },
  carts: {
    deleteDaysAfterLastModification: 90,
    countryTaxRateFallbackEnabled: false,
  },
};

export function renderWithProviders(ui: React.ReactElement, extendedRenderOptions: ExtendedRenderOptions = {}) {
  const {
    preloadedState = {
      project: mockShop,
    },
    // Automatically create a store instance if no store was passed in
    store = initStore(preloadedState.project as Project),
    ...renderOptions
  } = extendedRenderOptions;

  const Wrapper = ({ children }: PropsWithChildren) => (
    <MemoryRouter>
      <LocalizationProvider dateAdapter={AdapterDayjs}>
        <Provider store={store}>{children}</Provider>
      </LocalizationProvider>
    </MemoryRouter>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
