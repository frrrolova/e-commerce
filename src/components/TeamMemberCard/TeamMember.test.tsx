import { renderWithProviders } from '@/utils/test-utils';
import { screen } from '@testing-library/react';
import TeamMemberCard from './TeamMemberCard';
import Placeholder from '/images/catalog/placeholder_plant.png';

jest.mock('@client/client', () => {
  return null;
});

const member = {
  role: 'Developer',
  name: 'Vasya',
  bio: 'Lorem ipsum dolor sit amet',
  photo: Placeholder,
  git: '',
  contributions: ['CommerceTools Project Setup'],
};

describe('Team Member Card', () => {
  test('Data in document is correct', async () => {
    renderWithProviders(<TeamMemberCard teamMember={member} />);
    const name = screen.getByText('Vasya');
    expect(name).toBeInTheDocument();
  });

  test('Data in document is correct', async () => {
    renderWithProviders(<TeamMemberCard teamMember={member} />);
    const role = screen.getByText('Developer');
    expect(role).toBeInTheDocument();
  });
});
