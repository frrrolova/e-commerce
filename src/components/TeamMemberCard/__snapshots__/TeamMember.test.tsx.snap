// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Member Card Data in document is correct 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-18v18mo-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-x8ntnm"
          >
            <header
              class="MuiBox-root css-hkmrg5"
            >
              <h2
                class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
              >
                Vasya
              </h2>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-136cz29-MuiTypography-root-MuiLink-root"
                href=""
              >
                Github
              </a>
            </header>
            <div
              class="MuiBox-root css-ig46ls"
            >
              <div
                class="MuiBox-root css-1szi7j8"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ncmvc0-MuiTypography-root"
              >
                Developer
              </p>
              <h4
                class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
              >
                Bio
              </h4>
              <h5
                class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
              >
                Lorem ipsum dolor sit amet
              </h5>
            </div>
            <footer
              class="MuiBox-root css-0"
            >
              <h3
                class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
              >
                Contributions
              </h3>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiBox-root css-hy5eh7"
                >
                  ✅ 
                  CommerceTools Project Setup
                </div>
              </div>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-18v18mo-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-x8ntnm"
        >
          <header
            class="MuiBox-root css-hkmrg5"
          >
            <h2
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              Vasya
            </h2>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-136cz29-MuiTypography-root-MuiLink-root"
              href=""
            >
              Github
            </a>
          </header>
          <div
            class="MuiBox-root css-ig46ls"
          >
            <div
              class="MuiBox-root css-1szi7j8"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-1ncmvc0-MuiTypography-root"
            >
              Developer
            </p>
            <h4
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Bio
            </h4>
            <h5
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet
            </h5>
          </div>
          <footer
            class="MuiBox-root css-0"
          >
            <h3
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Contributions
            </h3>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                CommerceTools Project Setup
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Team Member Card Data in document is correct 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-18v18mo-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-x8ntnm"
          >
            <header
              class="MuiBox-root css-hkmrg5"
            >
              <h2
                class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
              >
                Vasya
              </h2>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-136cz29-MuiTypography-root-MuiLink-root"
                href=""
              >
                Github
              </a>
            </header>
            <div
              class="MuiBox-root css-ig46ls"
            >
              <div
                class="MuiBox-root css-1szi7j8"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-1ncmvc0-MuiTypography-root"
              >
                Developer
              </p>
              <h4
                class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
              >
                Bio
              </h4>
              <h5
                class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
              >
                Lorem ipsum dolor sit amet
              </h5>
            </div>
            <footer
              class="MuiBox-root css-0"
            >
              <h3
                class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
              >
                Contributions
              </h3>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiBox-root css-hy5eh7"
                >
                  ✅ 
                  CommerceTools Project Setup
                </div>
              </div>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-18v18mo-MuiPaper-root"
      >
        <div
          class="MuiBox-root css-x8ntnm"
        >
          <header
            class="MuiBox-root css-hkmrg5"
          >
            <h2
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              Vasya
            </h2>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-136cz29-MuiTypography-root-MuiLink-root"
              href=""
            >
              Github
            </a>
          </header>
          <div
            class="MuiBox-root css-ig46ls"
          >
            <div
              class="MuiBox-root css-1szi7j8"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 css-1ncmvc0-MuiTypography-root"
            >
              Developer
            </p>
            <h4
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Bio
            </h4>
            <h5
              class="MuiTypography-root MuiTypography-body1 css-mzzmr3-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet
            </h5>
          </div>
          <footer
            class="MuiBox-root css-0"
          >
            <h3
              class="MuiTypography-root MuiTypography-body1 css-1fqel3p-MuiTypography-root"
            >
              Contributions
            </h3>
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-hy5eh7"
              >
                ✅ 
                CommerceTools Project Setup
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
